src/test/java/unit/com/mathworks/ci/actions/RunMatlabBuildActionTest.java

```diff
@@ -0,0 +1,170 @@
+package com.mathworks.ci.actions;
+
+/**
+ * Copyright 2024, The MathWorks Inc.
+ *
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Files;
+
+import org.junit.Test;
+import org.junit.Before;
+import org.junit.runner.RunWith;
+import static org.junit.Assert.*;
+
+import org.mockito.Mock;
+import org.mockito.InOrder;
+import org.mockito.junit.MockitoJUnitRunner;
+import static org.mockito.Mockito.*;
+
+import hudson.FilePath;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+
+import com.mathworks.ci.BuildArtifactAction;
+import com.mathworks.ci.BuildConsoleAnnotator;
+import com.mathworks.ci.MatlabExecutionException;
+import com.mathworks.ci.utilities.MatlabCommandRunner;
+import com.mathworks.ci.parameters.BuildActionParameters;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RunMatlabBuildActionTest {
+    @Mock BuildActionParameters params;
+    @Mock BuildConsoleAnnotator annotator;
+    @Mock MatlabCommandRunner runner;
+    @Mock TaskListener listener;
+    @Mock PrintStream out;
+    @Mock Run build;
+
+    @Mock FilePath tempFolder;
+
+    private boolean setup = false;
+    private RunMatlabBuildAction action;
+
+    @Before
```

Comment:
You can use BeforeClass which will be run once before all the tests. https://junit.org/junit4/javadoc/4.12/org/junit/BeforeClass.html

Same for all other tests.

Reply:
Oh yeah, using `BeforeClass` requires using a `static` method, which would mean that all of the other properties on the class set in `init` also have to be `static`. I generally prefer to stay away from static variables if possible, but in this case I don't think switching the test class properties to be static would have any serious negative (or positivie) impact. Overall though, I still lean slightly towards avoiding it due to some negative past experiences with having to reason about static variables.

Let me know if you feel strongly and I can get that switched over if it's a dealbreaker for you, but using `@Before` instead of `@BeforeClass` was definitely a conscious choice if that changes your input at all. I'd rather be setting encapsulated data on the test class object itself than having to set static data in a static init method.

Reply:
I think it's fine to use @Before. Can we add a comment regarding the same?

Reply:
Yep, I'll get a comment added.